include/ulight/impl/lang/xml_chars.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef ULIGHT_XML_CHARS_HPP
+#define ULIGHT_XML_CHARS_HPP
+
+#include "ulight/impl/ascii_chars.hpp"
+
+namespace ulight::xml {
+
+/// @brief returns true if c is whitespace according to the XML standard
+[[nodiscard]]
+constexpr bool is_xml_whitespace_character(char32_t c)
+{
+
+    return c == U'\u0009' //
+        || c == U'\u0020' //
+        || c == U'\u000d' //
+        || c == U'\u000a';
+}
+
```

Comment:
```suggestion

constexpr bool is_xml_name_start_character(char8_t c) = delete;

```
Whenever the function only works with `char32_t`, there should be such an extra overload for `char8_t` because otherwise, it's easily to accidentally convert `char8_t` to `char32_t` and thus perform a character test that cannot be done with code units alone (but requires whole code points).